include/feature_lib.h

```diff
@@ -0,0 +1,62 @@
+#ifndef FEATUREDETECT_FEATURELIB_H_
+#define FEATUREDETECT_FEATURELIB_H_
+
+#include <vector>
+#include <opencv2/features2d.hpp>
+
+namespace featurelib {
+
+struct KeyPoint {
+  float x;
+  float y;
+  float size;
+  float angle;
+  float strength;
+};
+
+struct DetectionResult {
+  std::vector<KeyPoint> keypoints;
+  std::vector<std::vector<uint8_t>> descriptors;
+};
+
+class FeatureDetector {
+ public:
+  virtual DetectionResult detect(const std::vector<std::uint8_t> &input_vector) const = 0;
+
+  void setHeight(int height) {
+    mask.reshape(1, std::vector<int>{height, width});
+  }
+
+  void setWidth(int width) {
+    mask.reshape(1, std::vector<int>{height, width});
```

Comment:
I think, you cannot use `reshape` here (and for the height too) as the number of elements changes. You need to create a new matrix or change the size of this one somehow.

Reply:
for example, when only the width increases and the height remains the same, you need to somehow add columns, so I will recreate it

Reply:
Yep. There is also `cv::resize`, but I think that recreating would be faster because this resizing involves interpolation